cli/cmd/version.go

```diff
@@ -37,7 +37,9 @@ func VersionCommand() *cobra.Command {
 		Short: "Show the Docker version information",
 		Args:  cobra.MaximumNArgs(0),
 		Run: func(cmd *cobra.Command, _ []string) {
-			runVersion(cmd)
+			if err := runVersion(cmd); err != nil {
```

Comment:
why explicitely exit here and not let the error be managed in main.go ?? (this will prevent metrics working on this command)

Reply:
Moby cli returns a non-zero exit code when it fails to retrieve the engine version. Compose CLI ignores it.